### PR TITLE
Add tooling to release packages

### PR DESCRIPTION
**Pull request by wardpenney**
- Add bin setup to install machine dependencies
  for release script
- Add node engine version to package 

Thanks to https://github.com/primer/primer for
the scripts!

### CODE REVIEW

File: bin/setup [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env sh
+
+# Set up Ruby dependencies via Bundler
```
Comment (tysongach) [inline bin/setup]: Update this comment?

File: bin/setup [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env sh
```
Comment (tysongach) [inline bin/setup]: Should this be `#!/bin/sh` to match the other scripts?

File: bin/setup [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+
```
Comment (tysongach) [inline bin/setup]: Should this script have `set -e` (exits if any subcommand fails), like the other scripts?

File: scripts/release [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
```
Comment (tysongach) [inline scripts/release]: Is there a reason this script and `try_release` are in a `scripts` directory, but `setup` is in `bin`? Do they have different situations/goals, or should they be combined if not?

Comment (wardpenney) [reply scripts/release]: Oh maybe yeah let's combine. I'm just do used to the `./bin/setup` thoughtbot pattern.

Comment (tysongach) [reply scripts/release]: Cool, yeah! I'm good with `bin`; it's much easier to type than `scripts`. But whichever is fine, really.

Comment (wardpenney) [reply scripts/release]: Ohh you mean the other way around? The `scripts` is a pretty strong npm convention cause the tool that runs them is called npm scripts.

Comment (wardpenney) [reply scripts/release]: But we don't have to follow npm. I'll move to bin.

Comment (tysongach) [reply scripts/release]: Ha, sorry, I could've been more clear.

Looking at the original diff, I just didn't understand why some scripts were in `bin` and others in `scripts`. They were all shell scripts, so I wondered if they should all go together. But if npm has a strong convention, that's fine to keep them separated.

File: bin/setup [hunk 5]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+set -e
+
+if ! command -v ja > /dev/null; then
+  brew install jq
```
Comment (tysongach) [inline bin/setup]: I assume this locks us into macOS? I don't think we should worry about it right now, but something to think about.

Comment (wardpenney) [reply bin/setup]: Yeah, `jq` itself is very portable so other OS's could install it. I just don't know how to put that into this script 🙈

Comment (tysongach) [reply bin/setup]: Probably React, right?
<br>
<br>
<br>
<br>
<br>
<br>
<br>
:trollface:

**Pull request merged**